### PR TITLE
Refactors in SchemaManager related classes

### PR DESCRIPTION
**Pull request by kolosovpetro**
## Description
Refactors in SchemaManager related classes

## Related issues
[issue #319].

## Testing
Existing tests are not changed and passed

## Semver Change ([docs](https://github.com/microsoft/healthcare-shared-components/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking

### CODE REVIEW

File: src/Microsoft.Health.SqlServer/Features/Schema/Manager/BaseSchemaRunner.cs [hunk 1]
```diff
@@ -69,21 +68,25 @@ public async Task EnsureInstanceSchemaRecordExistsAsync(CancellationToken cancel
...
-            .WaitAndRetryAsync(
-                retryCount: RetryAttempts,
-                sleepDurationProvider: (retryCount) => RetrySleepDuration,
-                onRetry: (exception, retryCount) =>
-                {
-                    _logger.LogWarning(exception, string.Format(Resources.RetryInstanceSchemaRecord, attempts++, RetryAttempts));
-                })
-            .ExecuteAsync(token => InstanceSchemaRecordCreatedAsync(token), cancellationToken);
+                .WaitAndRetryAsync(
+                    RetryAttempts,
```
Comment (jovinson-ms) [inline src/Microsoft.Health.SqlServer/Features/Schema/Manager/BaseSchemaRunner.cs:72]: Let's leave the named arguments to this method - it helps with readability.

File: src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaClient.cs [hunk 2]
```diff
@@ -35,90 +35,121 @@ public void SetUri(Uri uri)
...
 
         public async Task<CompatibleVersion> GetCompatibilityAsync(CancellationToken cancellationToken)
         {
-            HttpResponseMessage response = await _httpClient.GetAsync(KnownRoutes.RootedCompatibilityUri, cancellationToken).ConfigureAwait(false);
-            if (response.IsSuccessStatusCode)
+            HttpResponseMessage response = await _httpClient
+                .GetAsync(KnownRoutes.RootedCompatibilityUri, cancellationToken)
+                .ConfigureAwait(false);
+
+            if (!response.IsSuccessStatusCode)
```
Comment (jovinson-ms) [inline src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaClient.cs:90]: as much as possible, we prioritize handling successful conditionals - can you change this block back to the true case?

**Review by jovinson-ms — COMMENTED** [review]
@kolosovpetro, thanks so much for picking up this issue! I looked through your changes and everything looks good - just had a few small comments. Before you merge this PR, I want to test it against downstream projects to make sure things work smoothly. I'll try to get that done by the end of this week. Thanks again for contributing!

**Comment by kolosovpetro** [comment]
> @kolosovpetro, thanks so much for picking up this issue! I looked through your changes and everything looks good - just had a few small comments. Before you merge this PR, I want to test it against downstream projects to make sure things work smoothly. I'll try to get that done by the end of this week. Thanks again for contributing!

Thanks for the review, I will update this PR shortly

**Comment by jovinson-ms** [comment]
Thanks, @kolosovpetro for working through the conflicts - I should have mentioned that we made the switch to target .NET 5 and 6. Anyway, I've checked on the downstream users and made note of some small follow-up PRs we'll need to update the namespaces, and tomorrow I'll take one more pass through the latest changes in this PR.

**Comment by wsugarman** [comment]
Pull request is stale. Please feel free to re-open at any time.

**Pull request closed without merging**